File: 0001_0100/0026/0026.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    x1, y1, r1 = map(int, f.readline().split())
    x2, y2, r2 = map(int, f.readline().split())
    m = (x1 - x2)**2 + (y1 - y2)**2
    q.write("NO" if m > (r1 + r2)**2 or m < (r1 - r2)**2 else "YES")

File: 0001_0100/0026/0026.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var x1, y1, r1, x2, y2, r2 int
	fmt.Fscan(f, &x1, &y1, &r1, &x2, &y2, &r2)
	m := (x1-x2)*(x1-x2) + (y1-y2)*(y1-y2)
	s := "YES"
	if m > (r1+r2)*(r1+r2) || m < (r2-r1)*(r2-r1) {
		s = "NO"
	}
	fmt.Fprint(q, s)
}
